dawn/setup.py

```diff
@@ -18,23 +18,65 @@
 import subprocess
 import sys
 
+from glob import glob
 from distutils.version import LooseVersion
 from setuptools import setup, find_packages, Command, Extension
 from setuptools.command.build_ext import build_ext
 
-
+# Based on:
+#   https://github.com/navdeep-G/setup.py
+NAME = "dawn4py"
+DESCRIPTION = "High-level DSL toolchain for geophysical fluid dynamics models."
+URL = "https://github.com/MeteoSwiss-APN/dawn"
+EMAIL = "gridtools@cscs.com"
+AUTHOR = "MeteoSwiss / ETH Zurich / Vulcan"
+
+# Note that DAWN_DIR below may be the version pip copies over before building
 DAWN_DIR = os.path.join(os.path.dirname(__file__))
-DAWN_ABS_DIR = os.path.abspath(DAWN_DIR)
 BUILD_JOBS = 4
 
 # Select protobuf version
-# TODO: avoid parsing python files and adapt to new CMake
 with open(os.path.join(DAWN_DIR, "cmake", "FetchProtobuf.cmake"), "r") as f:
```

Comment:
This is hardcoding the python protobuf package version to be installed to the version selected in `FetchProtobuf.cmake`. This creates a potential problem if there is a valid protobuf version installed in the system with a different version which gets used by CMake to build the binaries. In that case, the python protobuf package and binary protobuf will have different versions.

A simple solution would be to always use the FetchProtobuf version when building the `python` target, which would also fix the other problem I was experiencing, when a too old version of protobuf installed in the system messes up with the CMake build configuration.